lib/Serialization/Serialization.cpp

```diff
@@ -1864,6 +1866,7 @@ void Serializer::writeCrossReference(const DeclContext *DC, uint32_t pathLen) {
   case DeclContextKind::Initializer:
   case DeclContextKind::TopLevelCodeDecl:
   case DeclContextKind::SerializedLocal:
+  case DeclContextKind::EnumElementDecl:
     llvm_unreachable("cannot cross-reference this context");
```

Comment:
If EnumElementDecls grow proper ParamDecls, I think you *will* need to cross-reference this context.